plugins/onepad/joystick.cpp

```diff
@@ -218,7 +218,7 @@ bool JoystickInfo::Init(int id)
 	// Keep the 4 hat buttons too (usb driver). (left pressure does not work with recent kernel). Moreover the pressure
 	// work sometime on half axis neg others time in fulll axis. So better keep them as button for the moment
 	u32 found_hack = devname.find(string("PLAYSTATION(R)3"));
-	if (found_hack != string::npos) {
+	if (found_hack != string::npos && (conf->options & PADOPTION_SIXAXIS_USB) == 0) {
```

Comment:
On my system, /dev/input/event13 (evdev interface) and /dev/input/js0 (joystick interface) correspond to the DS3 on my system. User/group ownership are both root.

With /dev/input/event13 at 0640 and /dev/input/js0 at 0644 - SDL2 detects no pad, SDL1 detects a 19 button, 27 axis pad

With /dev/input/event13 at 0666 and /dev/input/js0 at 0644 - Both SDL1 and SDL2 report a 19 button, 4 axis pad.

Can you check if your system has similar behaviour?

My conclusion is that SDL2 only uses the evdev interface, SDL1 uses the evdev interface but if that fails, it uses the joystick interface. And what most likely happened is there's been a change in the udev rules that enabled read/write access to the evdev interface for the DS3, which stops SDL1 from using the joystick interface.